aten/src/ATen/native/native_functions.yaml

```diff
@@ -2778,11 +2778,15 @@
     DefaultBackend: mvlgamma_
 
 - func: narrow_copy(Tensor self, int dim, int start, int length) -> Tensor
-  variants: method
+  variants: function, method
   dispatch:
     CPU, CUDA: narrow_copy_dense
     SparseCPU, SparseCUDA: narrow_copy_sparse
 
+- func: narrow_copy.out(Tensor self, int dim, int start, int length, *, Tensor(a!) out) -> Tensor(a!)
```

Comment:
Somehow I found my inline comment disappeared so redoing the comment :P 
You can simply leave `narrow_copy_dense` as it is and add a new `narrow_copy_cpu` with the cpu logic. And since the `narrow_copy_dense` actually works for all backends, you can simply update this section to (note no need to explicitly specify CUDA anymore)
```
CPU: narrow_copy_cpu
SparseCPU,......
Math: narrow_copy_dense
```

Reply:
It doesn't seem to work. Here are the error messages:

```
> RuntimeError: 0 INTERNAL ASSERT FAILED at "caffe2/aten/src/ATen/core/boxing/KernelFunction.cpp":27, please report a bug to PyTorch. aten::narrow_copy has kernels registered to both Math and a backend mapped to AutogradOther. This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). If it's intended to override Math kernel behavior, please open an issue to request a dedicated Autograd dispatch key for the backend.
Canonical state
~~~~~~~~~~~
name: aten::narrow_copy
schema: aten::narrow_copy(Tensor self, int dim, int start, int length) -> (Tensor)
debug: registered at buck-out/dev/gen/caffe2/aten/gen_aten=RegisterSchema.cpp/RegisterSchema.cpp:20
alias analysis kind: FROM_SCHEMA
CPU: registered at buck-out/dev/gen/caffe2/aten/gen_aten=RegisterCPU.cpp/RegisterCPU.cpp:5778 :: (Tensor _0, int _1, int _2, int _3) -> (Tensor _0) [ boxed unboxed ]
SparseCPU: registered at buck-out/dev/gen/caffe2/aten/gen_aten=RegisterSparseCPU.cpp/RegisterSparseCPU.cpp:543 :: (Tensor _0, int _1, int _2, int _3) -> (Tensor _0) [ boxed unboxed ]
SparseCUDA: registered at buck-out/dev/gen/caffe2/aten/gen_aten=RegisterSparseCUDA.cpp/RegisterSparseCUDA.cpp:639 :: (Tensor _0, int _1, int _2, int _3) -> (Tensor _0) [ boxed unboxed ]
Tracer: registered at buck-out/dev/gen/caffe2/generate-code/autograd/generated/TraceType_3.cpp:10432 :: (Tensor _0, int _1, int _2, int _3) -> (Tensor _0) [ boxed unboxed ]
Autograd[alias]: registered at buck-out/dev/gen/caffe2/generate-code/autograd/generated/VariableType_3.cpp:9879 :: (Tensor _0, int _1, int _2, int _3) -> (Tensor _0) [ boxed unboxed ]
Math[alias]: registered at buck-out/dev/gen/caffe2/aten/gen_aten=RegisterMath.cpp/RegisterMath.cpp:5590 :: (Tensor _0, int _1, int _2, int _3) -> (Tensor _0) [ boxed unboxed ]
```

Reply:
Ah I see, it's because currently no grad formula is implemented for `narrow_copy`. 
```
In [2]:  a = torch.rand(3, 3, requires_grad=True)

In [3]: b = a.narrow_copy(0, 0, 1)

In [4]: b
Out[4]: tensor([[0.6334, 0.4079, 0.6572]], grad_fn=<NotImplemented>)
```
Changing `Math` to `DefaultBackend` should fix it! :D